synthesis/power_performance_area.proto

```diff
@@ -118,16 +118,26 @@ message Performance {
   // Approximate invariant:
   //   clock_period_ps - critical_path_ps = setup_wns_ps
 
-  // If negative, not enough time to have data ready for DFF to take data, i.e.
-  // the combinational path that exceeds that time the most.
+  // A negative value indicates that the data is not available in time at the
+  // endpoint to properly sample the value. A zero value indicates
+  // that timing is met - positive values are atypical but can be used to report
+  // margin. These values are the worst of in2reg, reg2reg, reg2out, and in2out.
   optional sint32 setup_wns_ps = 3;
   optional sint64 setup_tns_ps = 4;
 
-  // If negative, new data arrives too early for the DFF to ingest
-  // the previous one reliably.
+  // A negative value indicates that new data arrives too early for the
+  // endpoint to properly sample the previous value. A zero value indicates
+  // that timing is met - positive values are atypical but can be used to report
+  // margin. These values are the worst of in2reg, reg2reg, reg2out, and in2out.
   optional sint32 hold_wns_ps = 5;
   optional sint64 hold_tns_ps = 6;
 
+  // Additional timing breakdowns for in2reg, reg2reg, reg2out, and in2out.
```

Comment:
Maybe worth spelling out in and out mean block input to first register and last register to block output.

Reply:
Good point, added those.